src/components/shared/Btn.vue

```diff
@@ -0,0 +1,49 @@
+<template>
+  <button :class="btnClass()" @click="click"><slot></slot></button>
+</template>
+<script>
+export default {
+  props: {
+    type: { type: String, default: 'primary' },
+    size: { type: String, default: 'medium' },
+  },
+  methods: {
+    click: function click(e) {
+      this.$emit('click', e);
+    },
+    btnClass: function btnClass() {
```

Comment:
isso aqui deveria estar no `computed`

Reply:
this is not needed any way. I have tested it. Tks